usaspending_api/references/v2/views/location_autocomplete.py

```diff
@@ -0,0 +1,307 @@
+
+models = [
+    {"key": "search_text", "name": "search_text", "type": "text", "text_type": "search", "optional": False},
+    {"key": "limit", "name": "limit", "type": "integer", "max": 500, "optional": True, "default": 10},
+]
+
+
+@validate_post_request(models)
+class LocationAutocompleteViewSet(APIView):
+    """
+    This end point returns a list of locations from Elasticsearch that match the given search_text value.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/autocomplete/location.md"
+
+    @cache_response()
+    def post(self, request):
+        es_results: ES_Response = self._query_elasticsearch(request.data["search_text"], request.data["limit"])
+
+        if len(es_results) > 0:
+            results = {}
+
+            country_results = self._format_country_results(
+                list(filter(lambda x: "country_name" in dir(x.meta.highlight), es_results))
+            )
+            state_results = self._format_state_results(
+                list(filter(lambda x: "state_name" in dir(x.meta.highlight), es_results))
+            )
+            city_results = self._format_city_results(
+                list(filter(lambda x: "cities" in dir(x.meta.highlight), es_results))
+            )
+            county_results = self._format_county_results(
+                list(filter(lambda x: "counties" in dir(x.meta.highlight), es_results))
+            )
+            zip_code_results = self._format_zip_code_results(
+                list(filter(lambda x: "zip_codes" in dir(x.meta.highlight), es_results))
+            )
+            current_cd_results = self._format_current_cd_results(
+                list(filter(lambda x: "current_congressional_districts" in dir(x.meta.highlight), es_results))
+            )
+            original_cd_results = self._format_original_cd_results(
+                list(filter(lambda x: "original_congressional_districts" in dir(x.meta.highlight), es_results))
+            )
+
+            if country_results is not None:
+                results["countries"] = country_results
+            if state_results is not None:
+                results["states"] = state_results
+            if city_results is not None:
+                results["cities"] = city_results
+            if county_results is not None:
+                results["counties"] = county_results
+            if zip_code_results is not None:
+                results["zip_codes"] = zip_code_results
+            if original_cd_results is not None:
+                results["districts_original"] = original_cd_results
+            if current_cd_results is not None:
+                results["districts_current"] = current_cd_results
+
+            # Account for cases where there are multiple results in a single ES document
+            results_length = sum(len(x) for x in results.values())
+
+            return Response(OrderedDict([("count", results_length), ("results", results), ("messages", [""])]))
```

Comment:
Just noticed the CodeClimate errors. Kind of annoying because it doesn't seem complex to me. I wonder if limiting the method to a single exit point (combining these returns) would knock it down a point? If not, let's force it through.

Reply:
Let me try combining those `return` statements. I fought with the complexity constantly with this endpoint and I think it's purely due to the number of different cases (7 to be exact) we have to cover with this endpoint.